persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -247,13 +258,26 @@ protected void extraStopSteps() {
       }
    }
 
+   @Override
+   public CompletionStage<Long> approximateSize(IntSet segments) {
+      // The size method ignores segments, see ISPN-13636
+      return size(segments).thenApply(totalSize -> {
```

Comment:
```suggestion
      return size(segments);
```
I don't see any reason to do the retainAll here. The segments should be handled in the size method directly.

Reply:
See the comment on line 263 :D

Reply:
I saw it, but there is no reason to do all this extra code IMO. The size method should be the one pushing this completely. The size method is already broken if you want to get approximateSize more accurate I would just say put this code in size is my point.

Reply:
I see what you mean now. 

At the time I thought you were working on a fix for `size()` and I didn't want to mess with it, but nowadays I'm thinking maybe the proper fix for ISPN-13636 is to remove the `segments` parameter from `NonBlockingStore.size()`.

The method is only really used in one place, `PersistenceManagerImpl.size(Predicate)`, which always calls it with the full set of segments, so it would be wasteful to write e.g. a SQL query that counts the rows for a set of segments that will only be used in the tests.

Reply:
Well the intent of `size(IntSet)` on a store is to eventually optimize the Cache.size() when it has a store without passivation so it doesn't need to even ask in memory and just asks the store, but that hasn't been done yet.

Reply:
Fair enough, I moved the approximation to `size()`